### DISCUSSION TITLE
Providing username suggestions

### DISCUSSION THREAD

**Discussion by teolemon** [openfoodfacts/openfoodfacts-auth#116] [Ideas]
Could we provide available  username suggestions ?
Asynchronous-Antilope-75 or https://www.geeksforgeeks.org/how-to-generate-a-unique-username-using-python/ ?
It was reported during the Open Prices call as a blocker for signup (people lacking imagination, and the username being already taken)
@hangy @raphael0202 @stephanegigandet

**Comment by hangy** [comment]
I think that should be possible. Add some custom JS code to [register](https://github.com/openfoodfacts/openfoodfacts-auth/blob/fff7aeb7d16dc2a9ca2284e7f426e5247ec74994/theme/keycloak.v2/login/register.ftl#L64), and use the [API](https://stackoverflow.com/a/52728664/11963) to check if a name already exists.
